presto-orc/src/main/java/com/facebook/presto/orc/metadata/Footer.java

```diff
@@ -36,8 +36,17 @@
     private final List<ColumnStatistics> fileStats;
     private final Map<String, Slice> userMetadata;
     private final Optional<DwrfEncryption> encryption;
+    private final Optional<List<Integer>> dwrfStripeCacheOffsets;
 
-    public Footer(long numberOfRows, int rowsInRowGroup, List<StripeInformation> stripes, List<OrcType> types, List<ColumnStatistics> fileStats, Map<String, Slice> userMetadata, Optional<DwrfEncryption> encryption)
+    public Footer(
+            long numberOfRows,
+            int rowsInRowGroup,
+            List<StripeInformation> stripes,
+            List<OrcType> types,
+            List<ColumnStatistics> fileStats,
+            Map<String, Slice> userMetadata,
+            Optional<DwrfEncryption> encryption,
+            Optional<List<Integer>> dwrfStripeCacheOffsets)
```

Comment:
Should the footer contain Optional<StripeCacheInfo> object. 
StripeCacheInfo class should contain 3 fields length, mode and offsets. 

Now the footer directly contain the offsets.